### PR TITLE
fix: rename next to remix

### PR DESCRIPTION
**Pull request by jacobparis**
when remix was forked from next it looks like you forgot to rename some things, this PR finishes renaming them to Remix

### CODE REVIEW

File: templates/cloudflare/tsconfig.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
     "isolatedModules": true,
     "esModuleInterop": true,
     "jsx": "react-jsx",
-    "module": "ESNext",
+    "module": "ESRemix",
```
Comment (remorses) [inline templates/cloudflare/tsconfig.json:16]: lol

**Comment by valtism** [comment]
LGTM 👍🏻

File: CHANGELOG.md [hunk 2]
```diff
@@ -275,7 +275,7 @@ With this change:
   - ✅ `return { data: "whatever" };`
 - If you _were_ returning a custom `status` or `headers` via `json`/`defer`:
   - We've added a new API-compatible [`unstable_data`](https://remix.run/docs/utils/data) utility that will let you send back `status`/`headers` alongside your raw data without having to encode it into a `Response`
-- We will be removing both `json` and `defer` in the next major version, but both _should_ still work in Single Fetch in v2 to allow for incremental adoption of the new behavior
+- We will be removing both `json` and `defer` in the remix major version, but both _should_ still work in Single Fetch in v2 to allow for incremental adoption of the new behavior
```
Comment (mmkal) [inline CHANGELOG.md:278]: Uhh I think this one should stay as next

Comment (remorses) [reply CHANGELOG.md:278]: Good catch, we are lucky you reviewed before it was merged

File: docs/discussion/pending-ui.md [hunk 3]
```diff
@@ -79,7 +79,7 @@ export function ProjectList({ projects }) {
 }
 ```
 
-Or add a spinner next to it by inspecting params:
+Or add a spinner remix to it by inspecting params:
```
Comment (amatzen) [inline docs/discussion/pending-ui.md:82]: 😭

File: CHANGELOG.md [hunk 4]
```diff
@@ -263,7 +263,7 @@ We found that the `future.unstable_fogOfWar` flag name could be a bit confusing
...
 
-However, the realization/alignment between Michael and Ryan on the recent [roadmap planning](https://www.youtube.com/watch?v=f5z_axCofW0) made us realize that the original assumption was incorrect. `middleware` won't need a `response` stub - as users can just mutate the `Response` they get from `await next()` directly.
+However, the realization/alignment between Michael and Ryan on the recent [roadmap planning](https://www.youtube.com/watch?v=f5z_axCofW0) made us realize that the original assumption was incorrect. `middleware` won't need a `response` stub - as users can just mutate the `Response` they get from `await remix()` directly.
```
Comment (Thanaen) [inline CHANGELOG.md:266]: I'm not sure we should rename "next" to "remix" here 😄

File: CHANGELOG.md (hunk 2 above)
Comment (Hammster) [inline CHANGELOG.md:278]: That looks like a search and replace mistake

File: DEVELOPMENT.md [hunk 5]
```diff
@@ -93,9 +93,9 @@ You may need to make changes to a pre-release prior to publishing a final stable
 - Once the release is published:
   - Pull the latest `release-*` branch containing the PR you just merged
   - Merge the `release-*` branch into `main` **using a non-fast-forward merge** and push it up to GitHub
-    - `git checkout main; git merge --no-ff release-next`
```
Comment (Hammster) [inline DEVELOPMENT.md:96]: Next in here is also not referring to the framework name

File: decisions/0005-remixing-react-router.md [hunk 6]
```diff
@@ -30,7 +30,7 @@ In [Remixing React Router][remixing router], Ryan gives an overview of the work
 
 ### Move the bulk of logic to a framework-agnostic router
 
-Thankfully this decision was sort of already made by Ryan. Maybe a surprise to some, maybe not, the current transition manager doesn't import or reference `react` or `react-router` a single time. This is by design because the logic being handled has nothing to do with how to render the UI layer. It's all about "what route am I on?", "what route am I going to?", "how do I load data for the next route?", "how do I interrupt ongoing navigations?" etc. None of these decisions actually _care_ about how the route and its data will eventually be rendered. Instead, the router simply needs to know whether given routes _have_ components and/or error boundaries - but it doesn't need to know about them or how to render them.
```
Comment (Hammster) [inline decisions/0005-remixing-react-router.md:33]: One more rename issue

**Review by Hammster — COMMENTED** [review]
Generally there are a lot of issues where "next" as a word was replaced with remix, the PR should be done with more care on these issues. Also changelogs should not be tempered with.

**Comment by xHomu** [comment]
LGTM!

File: CHANGELOG.md (hunk 2 above)
Comment (jacobparis) [reply CHANGELOG.md:278]: no this looks right to me, the remix major version

File: DEVELOPMENT.md (hunk 5 above)
Comment (jacobparis) [reply DEVELOPMENT.md:96]: we're trying to release-remix not release next js

File: docs/discussion/pending-ui.md (hunk 3 above)
Comment (paul-phan) [reply docs/discussion/pending-ui.md:82]: LGTM! 😆

**Pull request closed without merging**